salt/modules/cmdmod.py

```diff
@@ -734,6 +728,7 @@ def _run(
 
     if (
         python_shell is not True
+        and shell is not None
         and not salt.utils.platform.is_windows()
```

Comment:
You updated `utils.shlex_split` to work with windows, so I think this `and not salt.utils.platform.is_windows()` should be removed.